execution/limit.go

```diff
@@ -68,19 +71,32 @@ func (node *LimitedStream) Close() error {
 }
 
 func (node *LimitedStream) Next(ctx context.Context) (*Record, error) {
-	if node.limit > 0 {
+	tx := storage.GetStateTransactionFromContext(ctx)
+	limitState := storage.NewValueState(tx.WithPrefix(node.streamID.AsPrefix()).WithPrefix(limitPrefix))
+
+	var limit octosql.Value
+	err := limitState.Get(&limit)
+	if err != nil {
+		return nil, errors.Wrap(err, "couldn't get limit state")
+	}
+
+	if limit.AsInt() > 0 {
 		record, err := node.rs.Next(ctx)
 		if err != nil {
 			if err == ErrEndOfStream {
-				node.limit = 0
+				limit = octosql.MakeInt(0)
 				return nil, ErrEndOfStream
 			}
 			return nil, errors.Wrap(err, "LimitedStream: couldn't get record")
 		}
-		node.limit--
-		if node.limit == 0 {
+		limit = octosql.MakeInt(limit.AsInt() - 1)
+		if limit.AsInt() == 0 {
 			node.rs.Close()
 		}
+		err = limitState.Set(&limit)
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't set new limit state")
+		}
 		return record, nil
 	}
 
```

Comment:
I think it would improve our coherence of code if we agreed upon when is it appropriate to use multiple AsType functions, and when is it better to just define a value and use it everywhere. Here you use the limit.AsInt() function three times, so maybe it would be better to just define "limitValue := limit.AsInt()" and then use it everywhere? This is more about coding standards rather than optimizing the code, because this change is miniscule when it comes to performance

Reply:
This is definitely a case where a helper variable should be used. If it's used 3 times. I'll change it.